Automattic/wp-calypso
# A4A: Add support for archiving referral orders

Implements a feature that allows agencies to archive referral orders, helping to clean up their Referrals dashboard, which can become cluttered with incomplete orders.

Related to https://github.com/Automattic/automattic-for-agencies-dev/issues/849

## Proposed Changes

* Introduces a new Archive action in the Actions dropdown of the Referrals section table.
* Adds a dedicated Archived page within the Referrals section to view archived referral orders.
* Implements a confirmation modal to ensure users genuinely want to archive the referral.

## Testing Instructions

* Open the live link
* Login with the `a4ademo` account.
* Go to Referrals
* Click on the "actions" for a Pending referral.
* Click Archive. Confirm you get a modal.
* Click Archive on the modal. Confirm the Referral disappears from the list.
* Go to Referrals > Archived. Confirm you see the referral you just archived.

<img width="387" alt="Screenshot 2024-12-03 at 15 04 21" src="https://github.com/user-attachments/assets/1c10a7ec-c2a0-4672-889b-b14e598e5989">

<img width="255" alt="Screenshot 2024-12-03 at 15 05 10" src="https://github.com/user-attachments/assets/82ba7161-530c-410c-8d99-19ec7e8a175d">
<img width="950" alt="Screenshot 2024-12-03 at 15 04 28" src="https://github.com/user-attachments/assets/9c58d9de-fe25-4bf6-91f0-586fc3efde23">

## Pre-merge Checklist
